ClientLibrary/lib/src/main/java/com/microsoft/projectoxford/face/rest/WebServiceRequest.java

```diff
@@ -96,150 +87,156 @@ public Object request(String url, RequestMethod method, Map<String, Object> data
     }
 
     private Object get(String url) throws ClientException, IOException {
-        HttpGet request = new HttpGet(url);
-        request.setHeader(HEADER_KEY, mSubscriptionKey);
 
-        HttpResponse response = this.mClient.execute(request);
-        int statusCode = response.getStatusLine().getStatusCode();
-        if (statusCode == HttpStatus.SC_OK) {
-            return readInput(response.getEntity().getContent());
+        Request request = new Request.Builder()
+                .url(url)
+                .get()
+                .header(HEADER_KEY, mSubscriptionKey)
+                .build();
+
+        Response response = this.mClient.newCall(request).execute();
+        if (response.isSuccessful()) {
+            return readInput(response);
         } else {
-            String json = readInput(response.getEntity().getContent());
+            String json = readInput(response);
             if (json != null) {
                 ServiceError error = mGson.fromJson(json, ServiceError.class);
                 if (error != null) {
                     throw new ClientException(error.error);
                 }
             }
 
-            throw new ClientException("Error executing GET request!", statusCode);
+            throw new ClientException("Error executing GET request!", response.code());
         }
     }
 
 
     private Object patch(String url, Map<String, Object> data, String contentType) throws ClientException, IOException {
-        HttpPatch request = new HttpPatch(url);
-        request.setHeader(HEADER_KEY, mSubscriptionKey);
+
         String json = mGson.toJson(data);
-        StringEntity entity = new StringEntity(json, "utf-8");
-        request.setEntity(entity);
-        request.setHeader(CONTENT_TYPE, APPLICATION_JSON);
-        HttpResponse response = mClient.execute(request);
-
-        int statusCode = response.getStatusLine().getStatusCode();
-        if (statusCode == HttpStatus.SC_OK) {
-            return readInput(response.getEntity().getContent());
+
+        Request.Builder builder = new Request.Builder()
+                .url(url)
+                .header(HEADER_KEY, mSubscriptionKey)
+                .patch(RequestBody.create(MediaType.get(APPLICATION_JSON), json))
+                .header(CONTENT_TYPE, APPLICATION_JSON);
```

Comment:
[Nit] Can we unify the usage of `header` and `addHeader` if they are of same usage? Thanks.

Reply:
sure